### PR TITLE
docs(typedoc): add doc strings for typedoc

### PR DESCRIPTION
**Pull request by cogoo**
- docs(ts doc): add tsdoc strings for encode url
- docs(ts doc): tsdoc strings for create qr
- docs(encode url tsdoc): add object param
- docs(create transaction tsdocs): add tsdoc strings for create transaction
- docs(tsdoc find transaction signature): tsdoc strings for find transaction signature
- docs(tsdoc parse url): tsdoc strings for parse url
- docs(tsdoc validate transaction signature): tsdoc strings for validate transaction signature
- build(typedoc config): typedoc config

### CODE REVIEW

File: core/src/createTransaction.ts [hunk 1]
```diff
@@ -15,11 +15,28 @@ export class CreateTransactionError extends Error {
...
+ * **Reference** implementation for wallet providers.
+ *
+ * @param connection - A connection to the cluster.
+ * @param payer - `PublicKey` of the payer.
+ * @param recipient - The **native SOL** address the payment should be made to.
+ * @param amount - The amount of SOL or SPL token that should be transferred.
+ * @param {CreateTransactionParams} createTransactionParams - Additional parameters
+ * @param createTransactionParams.splToken - The mint address of the SPL token.
+ * @param createTransactionParams.reference - An array of public keys used to identify this transaction.
+ * @param createTransactionParams.memo - Creates an additional transaction for the [Memo Program](https://spl.solana.com/memo)
```
Comment (jacobcreech) [inline core/src/createTransaction.ts:38]: It's not an additional transaction, but an additional instruction on the same transaction

File: core/src/encodeURL.ts [hunk 2]
```diff
@@ -3,18 +3,38 @@ import BigNumber from 'bignumber.js';
...
     amount?: BigNumber;
+    /** The mint address of the SPL token */
     splToken?: PublicKey;
+    /** An array of public keys used to identify this transaction */
     reference?: PublicKey | PublicKey[];
+    /** A label to be used by the wallet provider to identify this transaction */
     label?: string;
+    /** A message to be used by the wallet provider to identify this transaction */
     message?: string;
+    /** Creates an additional transaction for the [Memo Program](https://spl.solana.com/memo) */
```
Comment (jacobcreech) [inline core/src/encodeURL.ts:16]: Same comment here

File: core/src/encodeURL.ts [hunk 3]
```diff
@@ -3,18 +3,38 @@ import BigNumber from 'bignumber.js';
...
+ *
+ * @param {EncodeURLComponents} encodeURLParams
+ *
+ * @param encodeURLParams.recipient - The address the payment should be made to. It **must** be a native SOL address.
+ * @param encodeURLParams.amount - The amount of SOL or SPL token that should be transferred. It  is always interpreted to be a decimal number of "user" units. If `null` the user will be requested to enter an amount by the wallet provider.
+ * @param encodeURLParams.splToken - The mint address of the SPL token. If `null` the transaction will be for native SOL
+ * @param encodeURLParams.reference - An array of public keys used to identify this transaction. They are the **only** way you'll be able to ensure that the customer has completed this transaction and payment is complete.
+ * @param encodeURLParams.label - A label to be used by the wallet provider to identify this transaction; should be the merchant name
+ * @param encodeURLParams.message - A message to be used by the wallet provider to identify this transaction; should describe the transaction to the user
+ * @param encodeURLParams.memo - Creates an additional transaction for the [Memo Program](https://spl.solana.com/memo)
```
Comment (jacobcreech) [inline core/src/encodeURL.ts:36]: Same comment

File: core/src/encodeURL.ts [hunk 4]
```diff
@@ -3,18 +3,38 @@ import BigNumber from 'bignumber.js';
...
     recipient: PublicKey;
 }
 
+/**
+ * Encode params into URL
+ *
+ * @param {EncodeURLComponents} encodeURLParams
+ *
+ * @param encodeURLParams.recipient - The address the payment should be made to. It **must** be a native SOL address.
+ * @param encodeURLParams.amount - The amount of SOL or SPL token that should be transferred. It  is always interpreted to be a decimal number of "user" units. If `null` the user will be requested to enter an amount by the wallet provider.
```
Comment (jacobcreech) [inline core/src/encodeURL.ts:31]: What is meant by `user` units?

File: core/src/findTransactionSignature.ts [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
 
+/**
+ * Find a transaction signature
+ *
+ * @throws if signature can't be found
+ *
+ * @param connection - A connection to the cluster.
+ * @param reference - A `PublicKey` that was included as a reference in the transaction.
+ * @param {SignaturesForAddressOptions} options - Options for getSignaturesForAddress.
+ * @param finality - A subset of Commitment levels, which are at least optimistically confirmed
```
Comment (jacobcreech) [inline core/src/findTransactionSignature.ts:16]: Nit: A user needs to know what are the possible levels. Should link Finality from web3

File: core/src/parseURL.ts [hunk 6]
```diff
@@ -3,19 +3,34 @@ import BigNumber from 'bignumber.js';
...
     amount: BigNumber | undefined;
+    /** The mint address of the SPL token */
     splToken: PublicKey | undefined;
+    /** An array of public keys used to identify this transaction */
     reference: PublicKey[] | undefined;
+    /** A label to be used by the wallet provider to identify this transaction */
     label: string | undefined;
+    /** A message to be used by the wallet provider to identify this transaction */
     message: string | undefined;
+    /** Creates an additional transaction for the [Memo Program](https://spl.solana.com/memo) */
```
Comment (jacobcreech) [inline core/src/parseURL.ts:18]: Same comment

**Review by jacobcreech — COMMENTED** [review]
Added some comments -- Open to discussion!

File: core/src/createTransaction.ts (hunk 1 above)
Comment (cogoo) [inline core/src/createTransaction.ts:38]:
```suggestion
 * @param createTransactionParams.memo - Creates an additional instruction for the [Memo Program](https://spl.solana.com/memo)
```

File: core/src/encodeURL.ts (hunk 2 above)
Comment (cogoo) [inline core/src/encodeURL.ts:16]:
```suggestion
    /** Creates an additional instruction for the [Memo Program](https://spl.solana.com/memo) */
```

Comment (cogoo) [inline core/src/encodeURL.ts]:
```suggestion
    /** Creates an additional instruction for the [Memo Program](https://spl.solana.com/memo) */
```

File: core/src/encodeURL.ts [hunk 7]
```diff
@@ -3,18 +3,38 @@ import BigNumber from 'bignumber.js';
...
+ * @param encodeURLParams.splToken - The mint address of the SPL token. If `null` the transaction will be for native SOL
+ * @param encodeURLParams.reference - An array of public keys used to identify this transaction. They are the **only** way you'll be able to ensure that the customer has completed this transaction and payment is complete.
+ * @param encodeURLParams.label - A label to be used by the wallet provider to identify this transaction; should be the merchant name
+ * @param encodeURLParams.message - A message to be used by the wallet provider to identify this transaction; should describe the transaction to the user
+ * @param encodeURLParams.memo - Creates an additional transaction for the [Memo Program](https://spl.solana.com/memo)
```
Comment (cogoo) [inline core/src/encodeURL.ts:36]:
```suggestion
 * @param encodeURLParams.memo - Creates an additional instruction for the [Memo Program](https://spl.solana.com/memo)
```

File: core/src/parseURL.ts (hunk 6 above)
Comment (cogoo) [inline core/src/parseURL.ts:18]:
```suggestion
    /** Creates an additional instruction for the [Memo Program](https://spl.solana.com/memo) */
```

File: core/src/encodeURL.ts (hunk 4 above)
Comment (cogoo) [reply core/src/encodeURL.ts:31]: From the spec:

> The amount field is always interpreted to be a decimal number of "user" units. For SOL, that's SOL and not lamports. For tokens, uiAmountString and not amount (reference: Token Balances Structure)

**Pull request merged**